src/gui/sharelinkwidget.cpp

```diff
@@ -154,24 +154,25 @@ void ShareLinkWidget::setupUiOptions()
     permissionsGroup->setExclusive(true);
 
     if (_isFile) {
-        checked = perm & (SharePermissionRead & SharePermissionUpdate);
-        _allowEditingLinkAction = _linkContextMenu->addAction(tr("Allow Editing"));
+        checked = (perm & SharePermissionRead) && (perm & SharePermissionUpdate);
```

Comment:
Is it just me or `perm & (SharePermissionRead | SharePermissionUpdate)` would have been more readable? (and faster, even though it's not really a hot path).

This comment applies to the other similar one below.